client/src/components/RotateLandscape/RotateLandscape.js

```diff
@@ -1,43 +1,66 @@
+import { Fragment } from "react";
+import { connect } from "react-redux";
 import MediaQuery from "react-responsive";
+
 import "./RotateLandscape.scss";
-import text from "./RotateLandscape.yaml";
-import svgs from "./RotateLandscapeSVG.yaml";
+import { IconRotatePhone, IconExpandWindowWidth } from "../../icons/icons.js";
+
+import { rotate_landscape_off } from "../../InfoBase/AppState.js";
 import { create_text_maker_component } from "../misc_util_components";
-import { Fragment } from "react";
 
-const { TM } = create_text_maker_component(text);
+import text from "./RotateLandscape.yaml";
+
+const { TM, text_maker } = create_text_maker_component(text);
 
-export default class RotateLandscape extends React.Component {
+class LandscapeOverlay extends React.Component {
   render() {
-    const { children } = this.props;
+    const { children, rotate_landscape, rotate_landscape_off } = this.props;
 
     return (
-      <Fragment>
-        <MediaQuery maxDeviceWidth={567.98}>
-          <MediaQuery orientation="portrait">
-            <div className="overlay">
-              <TM k="rotate_text" el="h2" style={{ color: "white" }} />
-              <div
-                className="icon"
-                dangerouslySetInnerHTML={{ __html: svgs.rotate_icon }}
-              />
-            </div>
-          </MediaQuery>
-          <MediaQuery orientation="landscape">{children}</MediaQuery>
-        </MediaQuery>
-        <MediaQuery minDeviceWidth={568}>
-          <MediaQuery maxWidth={567.98}>
-            <div className="overlay">
-              <TM k="expand_width_text" el="h2" style={{ color: "white" }} />
-              <div
-                className="icon"
-                dangerouslySetInnerHTML={{ __html: svgs.expand_width_icon }}
-              />
-            </div>
-          </MediaQuery>
-          <MediaQuery minWidth={568}>{children}</MediaQuery>
-        </MediaQuery>
-      </Fragment>
+      <div style={{ position: "relative" }}>
+        {children}
+        {rotate_landscape ? (
```

Comment:
Minor things but this can be short circuited `rotate_landscape &&`
and also
```
              <div className="overlay">
                <TM k="expand_width_text" el="h2" style={{ color: "white" }} />
                <IconExpandWindowWidth width="50%" />
                <button
                  className="btn btn-ib-primary"
                  onClick={rotate_landscape_off}
                >
                  {text_maker("show_anyway")}
                </button>
              </div>
```
this is redundant so we can declare it in variable and just call it

Reply:
Also `rotate_landscape_off` name is a bit unintuitive for me. Something like `turn_landscape_overlay_off` would be better or `ignore_landscape_overlay`